src/ebusd/mqtthandler.cpp

```diff
@@ -151,6 +154,14 @@ static error_t mqtt_parse_opt(int key, char *arg, struct argp_state *state) {
     g_ignoreInvalidParams = true;
     break;
 
+  case O_CLID:  // --mqttclientid=clientid
+    if (arg == nullptr) {
```

Comment:
Just guessing, but maybe add 

    || arg[0] == 0

to check for 0 length, same as O_TOPI?